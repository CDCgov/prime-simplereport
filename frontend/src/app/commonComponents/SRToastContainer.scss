.Toastify {
  &__toast-container {
    max-width: 40rem;
    width: auto;
  }

  &__toast {
    padding: 0;
  }

  &__close-button {
    height: 1.5rem;
    margin-top: 0.5rem;
    margin-right: 0.5rem;
    opacity: 1;
    width: 1.5rem;

    &:hover {
      opacity: 0.3;
    }

    svg {
      width: 100%;
      height: 100%;
    }
  }

  .sr-alert {
    &--error {
      background-color: #f4e3db;
    }

    &--success {
      background-color: #ecf3ec;

      .usa-alert--success {
        outline: none;
      }
    }
  }

  .usa-alert {
    max-width: 37.5rem;
  }
}
