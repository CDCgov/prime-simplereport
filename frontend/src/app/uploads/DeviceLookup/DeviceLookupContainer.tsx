import {
  DeviceType,
  useGetDeviceTypesForLookupQuery,
} from "../../../generated/graphql";
import { LoadingCard } from "../../commonComponents/LoadingCard/LoadingCard";

import DeviceLookup from "./DeviceLookup";

const DeviceLookupContainer = () => {
  const { data: deviceTypeResults } = useGetDeviceTypesForLookupQuery({
    fetchPolicy: "no-cache",
  });

  if (deviceTypeResults) {
    const devices = Array.from(
      deviceTypeResults.deviceTypes.map(
        (devicesTypes) => devicesTypes as DeviceType
      )
    );

    return <DeviceLookup deviceOptions={devices} />;
  } else {
    return <LoadingCard message="Loading" />;
  }
};

export default DeviceLookupContainer;
