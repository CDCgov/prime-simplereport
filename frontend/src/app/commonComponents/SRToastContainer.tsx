import React from "react";
import { ToastContainer } from "react-toastify";

import "react-toastify/dist/ReactToastify.css";
import "./SRToastContainer.scss";

const SRToastContainer = () => {
  const autoClose = 5000;
  const closeButton = true;
  const hideProgressBar = true;
  const limit = 2;
  const position = "bottom-left";

  return (
    <ToastContainer
      autoClose={autoClose}
      closeButton={closeButton}
      hideProgressBar={hideProgressBar}
      limit={limit}
      position={position}
    />
  );
};

export default SRToastContainer;
