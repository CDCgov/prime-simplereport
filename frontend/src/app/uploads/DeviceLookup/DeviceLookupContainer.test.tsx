import { render, screen } from "@testing-library/react";

import SRToastContainer from "../../commonComponents/SRToastContainer";

import DeviceLookupContainer from "./DeviceLookupContainer";

jest.mock("../../../generated/graphql", () => {
  return {
    useGetDeviceTypesForLookupQuery: () => {
      return {
        data: {
          deviceTypes: [
            {
              internalId: "1",
              name: "Abbott Tomorrow",
              loincCode: "loinc-1",
              manufacturer: "Abbott",
              model: "A",
              swabTypes: [
                {
                  internalId: "1234",
                  name: "NOSE POKE",
                  typeCode: "4321-a",
                },
              ],
            },
          ],
        },
      };
    },
  };
});

describe("DeviceLookupContainer", () => {
  beforeEach(() => {
    render(
      <>
        <DeviceLookupContainer />
        <SRToastContainer />
      </>
    );
  });

  it("should render device lookup container", async () => {
    expect(screen.getByText("Device lookup")).toBeInTheDocument();
  });
});
