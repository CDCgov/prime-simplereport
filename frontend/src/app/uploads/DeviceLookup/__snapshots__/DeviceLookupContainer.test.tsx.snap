// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeviceLookupContainer should render device lookup container 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <main
        class="prime-home device-lookup-container"
      >
        <div
          class="grid-container"
        >
          <div
            class="grid-row"
          >
            <div
              class="prime-container card-container"
            >
              <div
                class="usa-card__header"
              >
                <div>
                  <h2>
                    Device lookup
                  </h2>
                </div>
                <div
                  style="display: flex; justify-content: center; align-items: center;"
                />
              </div>
              <div
                class="usa-card__body margin-top-1"
              >
                <div
                  class="grid-row grid-gap"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <strong>
                      Select device
                    </strong>
                    <p />
                    <div
                      class="usa-combo-box usa-combo-box__full-width"
                      data-enhanced="true"
                      data-testid="combo-box"
                    >
                      <select
                        aria-hidden="true"
                        class="usa-select usa-sr-only usa-combo-box__select"
                        data-testid="combo-box-select"
                        name="selectDevice"
                        tabindex="-1"
                      >
                        <option
                          value="1"
                        >
                          Abbott Tomorrow
                        </option>
                      </select>
                      <input
                        aria-activedescendant=""
                        aria-autocomplete="list"
                        aria-describedby="selectDevice--assistiveHint"
                        aria-expanded="false"
                        aria-owns="selectDevice--list"
                        autocapitalize="off"
                        autocomplete="off"
                        class="usa-combo-box__input"
                        data-testid="combo-box-input"
                        id="selectDevice"
                        role="combobox"
                        type="text"
                        value=""
                      />
                      <span
                        class="usa-combo-box__clear-input__wrapper"
                        tabindex="-1"
                      >
                        <button
                          aria-label="Clear the select contents"
                          class="usa-combo-box__clear-input"
                          data-testid="combo-box-clear-button"
                          hidden=""
                          type="button"
                        >
                           
                        </button>
                      </span>
                      <span
                        class="usa-combo-box__input-button-separator"
                      >
                         
                      </span>
                      <span
                        class="usa-combo-box__toggle-list__wrapper"
                        tabindex="-1"
                      >
                        <button
                          aria-label="Toggle the dropdown list"
                          class="usa-combo-box__toggle-list"
                          data-testid="combo-box-toggle"
                          tabindex="-1"
                          type="button"
                        >
                           
                        </button>
                      </span>
                      <ul
                        class="usa-combo-box__list"
                        data-testid="combo-box-option-list"
                        hidden=""
                        id="selectDevice--list"
                        role="listbox"
                        tabindex="-1"
                      >
                        <li
                          aria-posinset="1"
                          aria-selected="false"
                          aria-setsize="1"
                          class="usa-combo-box__list-option"
                          data-testid="combo-box-option-1"
                          data-value="1"
                          id="selectDevice--list--option-0"
                          role="option"
                          tabindex="-1"
                          value="1"
                        >
                          Abbott Tomorrow
                        </li>
                      </ul>
                      <div
                        class="usa-combo-box__status usa-sr-only"
                        role="status"
                      />
                      <span
                        class="usa-sr-only"
                        data-testid="combo-box-assistive-hint"
                        id="selectDevice--assistiveHint"
                      >
                        When autocomplete results are available use up and down arrows to review
           and enter to select. Touch device users, explore by touch or with swipe
           gestures.
                      </span>
                    </div>
                  </div>
                </div>
                <div
                  class="grid-row grid-gap"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <div
                      class="text-field"
                    >
                      <div
                        class="usa-form-group"
                      >
                        <label
                          class="usa-label"
                          for="1"
                        >
                          Equipment model name
                        </label>
                        <span
                          class="usa-hint"
                        >
                          <code>
                            equipment_model_name
                          </code>
                        </span>
                        <input
                          aria-disabled="true"
                          aria-required="false"
                          class="usa-input"
                          disabled=""
                          id="1"
                          name="Equipment model name"
                          type="text"
                          value=""
                        />
                      </div>
                      <div
                        class="copy-button-container copy-button-tooltip"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="grid-row grid-gap"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <div
                      class="text-field"
                    >
                      <div
                        class="usa-form-group"
                      >
                        <label
                          class="usa-label"
                          for="2"
                        >
                          Test performed code 
                        </label>
                        <span
                          class="usa-hint"
                        >
                          <code>
                            test_performed_code
                          </code>
                        </span>
                        <input
                          aria-disabled="true"
                          aria-required="false"
                          class="usa-input"
                          disabled=""
                          id="2"
                          name="Test performed code "
                          type="text"
                          value=""
                        />
                      </div>
                      <div
                        class="copy-button-container copy-button-tooltip"
                      />
                    </div>
                  </div>
                </div>
                <div
                  class="grid-row grid-gap"
                >
                  <div
                    class="tablet:grid-col"
                  >
                    <div
                      style="position: relative;"
                    >
                      <div
                        class="usa-form-group"
                      >
                        <label
                          class="usa-label"
                        >
                          Specimen Type
                        </label>
                        <span
                          class="usa-hint"
                        >
                          <code>
                            specimen_type
                          </code>
                        </span>
                        <table
                          class="usa-table"
                        >
                          <tbody />
                        </table>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
      <div
        class="Toastify"
      />
    </div>
  </body>,
  "container": <div>
    <main
      class="prime-home device-lookup-container"
    >
      <div
        class="grid-container"
      >
        <div
          class="grid-row"
        >
          <div
            class="prime-container card-container"
          >
            <div
              class="usa-card__header"
            >
              <div>
                <h2>
                  Device lookup
                </h2>
              </div>
              <div
                style="display: flex; justify-content: center; align-items: center;"
              />
            </div>
            <div
              class="usa-card__body margin-top-1"
            >
              <div
                class="grid-row grid-gap"
              >
                <div
                  class="tablet:grid-col"
                >
                  <strong>
                    Select device
                  </strong>
                  <p />
                  <div
                    class="usa-combo-box usa-combo-box__full-width"
                    data-enhanced="true"
                    data-testid="combo-box"
                  >
                    <select
                      aria-hidden="true"
                      class="usa-select usa-sr-only usa-combo-box__select"
                      data-testid="combo-box-select"
                      name="selectDevice"
                      tabindex="-1"
                    >
                      <option
                        value="1"
                      >
                        Abbott Tomorrow
                      </option>
                    </select>
                    <input
                      aria-activedescendant=""
                      aria-autocomplete="list"
                      aria-describedby="selectDevice--assistiveHint"
                      aria-expanded="false"
                      aria-owns="selectDevice--list"
                      autocapitalize="off"
                      autocomplete="off"
                      class="usa-combo-box__input"
                      data-testid="combo-box-input"
                      id="selectDevice"
                      role="combobox"
                      type="text"
                      value=""
                    />
                    <span
                      class="usa-combo-box__clear-input__wrapper"
                      tabindex="-1"
                    >
                      <button
                        aria-label="Clear the select contents"
                        class="usa-combo-box__clear-input"
                        data-testid="combo-box-clear-button"
                        hidden=""
                        type="button"
                      >
                         
                      </button>
                    </span>
                    <span
                      class="usa-combo-box__input-button-separator"
                    >
                       
                    </span>
                    <span
                      class="usa-combo-box__toggle-list__wrapper"
                      tabindex="-1"
                    >
                      <button
                        aria-label="Toggle the dropdown list"
                        class="usa-combo-box__toggle-list"
                        data-testid="combo-box-toggle"
                        tabindex="-1"
                        type="button"
                      >
                         
                      </button>
                    </span>
                    <ul
                      class="usa-combo-box__list"
                      data-testid="combo-box-option-list"
                      hidden=""
                      id="selectDevice--list"
                      role="listbox"
                      tabindex="-1"
                    >
                      <li
                        aria-posinset="1"
                        aria-selected="false"
                        aria-setsize="1"
                        class="usa-combo-box__list-option"
                        data-testid="combo-box-option-1"
                        data-value="1"
                        id="selectDevice--list--option-0"
                        role="option"
                        tabindex="-1"
                        value="1"
                      >
                        Abbott Tomorrow
                      </li>
                    </ul>
                    <div
                      class="usa-combo-box__status usa-sr-only"
                      role="status"
                    />
                    <span
                      class="usa-sr-only"
                      data-testid="combo-box-assistive-hint"
                      id="selectDevice--assistiveHint"
                    >
                      When autocomplete results are available use up and down arrows to review
           and enter to select. Touch device users, explore by touch or with swipe
           gestures.
                    </span>
                  </div>
                </div>
              </div>
              <div
                class="grid-row grid-gap"
              >
                <div
                  class="tablet:grid-col"
                >
                  <div
                    class="text-field"
                  >
                    <div
                      class="usa-form-group"
                    >
                      <label
                        class="usa-label"
                        for="1"
                      >
                        Equipment model name
                      </label>
                      <span
                        class="usa-hint"
                      >
                        <code>
                          equipment_model_name
                        </code>
                      </span>
                      <input
                        aria-disabled="true"
                        aria-required="false"
                        class="usa-input"
                        disabled=""
                        id="1"
                        name="Equipment model name"
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="copy-button-container copy-button-tooltip"
                    />
                  </div>
                </div>
              </div>
              <div
                class="grid-row grid-gap"
              >
                <div
                  class="tablet:grid-col"
                >
                  <div
                    class="text-field"
                  >
                    <div
                      class="usa-form-group"
                    >
                      <label
                        class="usa-label"
                        for="2"
                      >
                        Test performed code 
                      </label>
                      <span
                        class="usa-hint"
                      >
                        <code>
                          test_performed_code
                        </code>
                      </span>
                      <input
                        aria-disabled="true"
                        aria-required="false"
                        class="usa-input"
                        disabled=""
                        id="2"
                        name="Test performed code "
                        type="text"
                        value=""
                      />
                    </div>
                    <div
                      class="copy-button-container copy-button-tooltip"
                    />
                  </div>
                </div>
              </div>
              <div
                class="grid-row grid-gap"
              >
                <div
                  class="tablet:grid-col"
                >
                  <div
                    style="position: relative;"
                  >
                    <div
                      class="usa-form-group"
                    >
                      <label
                        class="usa-label"
                      >
                        Specimen Type
                      </label>
                      <span
                        class="usa-hint"
                      >
                        <code>
                          specimen_type
                        </code>
                      </span>
                      <table
                        class="usa-table"
                      >
                        <tbody />
                      </table>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
    <div
      class="Toastify"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
